app/components/common/textBlurb/textBlurbDirective.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * textBlurbDirective.js
+ *
+ * A text box with a heading, and a read more link that expands the box to show the full contents
+ **/
+(function() {
+  'use strict';
+  
+  textBlurbDirective.$inject = ['$window'];
+  function textBlurbDirective($window) {
+    var directive = {
+      restrict: 'E',
+      bindToController: {
+        title: '@',
+        body: '@',
+      },
+      link: link,
+      controller: function() {},
+      controllerAs: 'textBlurb',
+      template: require('./textBlurb.html')
+    };
+
+    return directive;
+
+    //////////////////////////////
+    function link(scope, element, attrs, textBlurb) {
+      textBlurb.isCollapsed = true;
+      
+      scope.$watch(function() {return textBlurb.body; }, refreshButtons);
+      angular.element($window).bind('resize', refreshButtons);
+
+      function refreshButtons() {
+        textBlurb.showButtons = shouldShowButtons();
+        scope.$digest(); //since you're outside of angular, you need this here to update the view 
```

Comment:
Oh. good to know. I'll test and make sure the window resize thing works. If
it doesn't, I'll put back the extra digest call. It seemed ok the other
day, but will look again.

On Wed, Aug 24, 2016 at 7:38 PM Aaron D Borden <notifications@github.com>
wrote:

> In app/components/common/textBlurb/textBlurbDirective.js
> <https://github.com/caciviclab/disclosure-frontend/pull/137#discussion_r76172455>
> :
>
> > +      controllerAs: 'textBlurb',
> > +      template: require('./textBlurb.html')
> > +    };
> > +
> > +    return directive;
> > +
> > +    //////////////////////////////
> > +    function link(scope, element, attrs, textBlurb) {
> > +      textBlurb.isCollapsed = true;
> > +
> > +      scope.$watch(function() {return textBlurb.body; }, refreshButtons);
> > +      angular.element($window).bind('resize', refreshButtons);
> > +
> > +      function refreshButtons() {
> > +        textBlurb.showButtons = shouldShowButtons();
> > +        scope.$digest(); //since you're outside of angular, you need this here to update the view
>
> This is because of the event handling. Anytime you're calling a handler
> from an event (typing, clicking, resizing) or doing something async, it's
> not going to be called within the angular context.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/caciviclab/disclosure-frontend/pull/137/files/7daa7d9270091f9d4868bf505cebe9f05b4d1999#r76172455>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AD9iQBp8N7BofOqN7KOcQqrwtjHd1oDHks5qjQASgaJpZM4JhQnK>
> .
>